src/lib/terkin/driver/vedirect_sensor.py

```diff
@@ -51,9 +54,34 @@ def __init__(self, settings=None):
         super().__init__(settings=settings)
 
         # Can be overwritten by ``.set_address()``.
-        self.device = settings['device']
+        self.device = settings["device"]
         self.timeout = 5
         self.driver = None
+        self.threading = bool(settings["async"])
+        self._raw_data= None
+        self._read_time = 0
+        self._shutdown = False
+
+    def __del__(self):
+        self.shutdown()
+
+    def shutdown(self):
+        self._shutdown = True
+
+    def _async_read(self):
+        """Read until shutdown"""
+        log.info("Starting async VEDirect thread")
+        while not self._shutdown:
+            try:
+                data = self.driver.read_data_single(timeout=5000)
```

Comment:
Maybe use a minimal sleep here in order not create a busy loop?
```suggestion
                data = self.driver.read_data_single(timeout=5000)
                time.sleep(0.001)
```